From b223ec6e6162271c5f6ffd9596ef8f7dfa129c77 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Mon, 26 Feb 2024 18:48:34 +0100
Subject: [PATCH 0680/1695] plat-stm32mp2: conf: enable HASH for stm32mp2
 platforms

Default enables stm32_hash and crypto framework for stm32mp2 platforms.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git aaadf5d3fab414d751ddc8614f1fabe004e36ba4 plat-stm32mp2: conf: enable HASH for stm32mp2 platforms]
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Change-Id: If45fd038e9b542404bc7398d590ba97096de5fa0
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/361539
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index 3cd009255..b168d7910 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -126,6 +126,7 @@ CFG_STM32_BSEC_WRITE ?= y
 CFG_STM32_EXTI ?= y
 CFG_STM32_FMC ?= y
 CFG_STM32_GPIO ?= y
+CFG_STM32_HASH ?= y
 CFG_STM32_HPDMA ?= y
 CFG_STM32_HSEM ?= y
 CFG_STM32_I2C ?= y
@@ -180,7 +181,7 @@ override CFG_STM32_CRYP := n
 endif # cfg-all-enabled, CFG_STM32_CRYP CFG_STM32_SAES
 
 # If any crypto driver is enabled, enable the crypto-framework layer
-ifeq ($(call cfg-one-enabled, CFG_STM32_SAES),y)
+ifeq ($(call cfg-one-enabled, CFG_STM32_HASH CFG_STM32_SAES),y)
 $(call force,CFG_STM32_CRYPTO_DRIVER,y)
 endif
 
-- 
2.47.2

